hack/lib/etcd.sh

```diff
@@ -28,7 +28,16 @@ kube::etcd::validate() {
   }
 
   # validate etcd port is free
-  if netstat -nat | grep "[\.:]${ETCD_PORT:?} .*LISTEN" >/dev/null 2>&1; then
+  local port_check_command
+  if which ss &> /dev/null && ss -Version | grep 'iproute2' &> /dev/null; then
```

Comment:
I'd use `command -v` as it's apparently kosher for implementations of `which` to [not set exit status](https://stackoverflow.com/a/677212) and look at loads of things other than installed binaries. On my machine, `which` is:

```
$ type which
which is aliased to `(alias; declare -f) | /usr/bin/which --tty-only --read-alias --read-functions --show-tilde --show-dot'
```